### DISCUSSION TITLE
Problem with starting and stopping recording in React

### DISCUSSION THREAD

**Discussion by rudzky** [katspaugh/wavesurfer.js#3544] [General]
Hi guys,
I have a problem with Record Plugin. My code is pretty simple:
```typescript
const handleRecord = () => {
  if (!record) return;
  if (record.isRecording()) {
    record.stopRecording();
    return;
  }

  record.startRecording();
};
```
```html
 <button onClick={handleRecord}>Start / Stop</button>
```
Recording start works well, but when I want to stop it, I need to click twice! When I click only once, a new record starts, but on the second click the recording finally finish, and as the result I get the the audio recorded from the very begining - not only this after second click. 

I noticed that `record.isRecording()` does not update to `true` after the recording started.

Any idea how to fix it?

**Comment by katspaugh** [comment]
Try reproducing it with a minimal example without React.
